server/controller/src/main/java/ai/starwhale/mlops/domain/system/agent/AgentCache.java

```diff
@@ -71,27 +72,31 @@ public void removeOfflineAgent(String agentSerialNumber) {
 
     public Agent nodeReport(Node node) {
         log.debug("node reported {}", node.getSerialNumber());
-        Agent agentReported = agentConverter.fromNode(node);
-        Agent residentAgent = agents.get(node.getSerialNumber());
-        if (null == residentAgent) {
-            agents.put(node.getSerialNumber(), agentReported);
-            agentReported = save(agentReported);
-            return new AgentUnModifiable(agentReported);
+        if (StringUtils.hasText(node.getSerialNumber())) {
+            Agent agentReported = agentConverter.fromNode(node);
+            Agent residentAgent = agents.get(node.getSerialNumber());
+            if (null == residentAgent) {
+                agents.put(node.getSerialNumber(), agentReported);
+                agentReported = save(agentReported);
+                return new AgentUnModifiable(agentReported);
+            } else {
+                residentAgent.setAgentVersion(agentReported.getAgentVersion());
+                residentAgent.setStatus(agentReported.getStatus());
+                residentAgent.setNodeInfo(agentReported.getNodeInfo());
+                residentAgent.setConnectTime(agentReported.getConnectTime());
+                return new AgentUnModifiable(residentAgent);
+            }
         } else {
-            residentAgent.setAgentVersion(agentReported.getAgentVersion());
-            residentAgent.setStatus(agentReported.getStatus());
-            residentAgent.setNodeInfo(agentReported.getNodeInfo());
-            residentAgent.setConnectTime(agentReported.getConnectTime());
-            return new AgentUnModifiable(residentAgent);
+            return null;
```

Comment:
The return value of the method is never used, remove the useless lines?

Reply:
done